docs/essentials/resources.md

```diff
@@ -0,0 +1,66 @@
+---
+id: essentials_resources
+title:  "Resource Management"
+---
+
+NIO offers several objects, primarily channels, that consume resources (such as operating system file handles) that need to be released when no longer needed. If channels are not closed reliably, resource leaks can occur, causing a number of issues.
+
+For this reason, ZIO-NIO provides such resources using the [ZIO `ZManaged` API][zio-managed]. For example, calling `FileChannel.open` will produce a value of `ZManaged[Blocking, IOException, FileChannel]`. The file will not actually be opened until the managed value is *used*. 
+
+## Simple Usage
+
+The most straight-forward way to use a managed resource is with the `use` method:
+
+```scala mdoc:silent
+import zio._
+import zio.blocking.Blocking
+import zio.nio.core.channels._
+import zio.nio.core.file.Path
+import java.io.IOException
+
+def useChannel(f: FileChannel): ZIO[Blocking, IOException, Unit] = ???
+
+val effect: ZIO[Blocking, IOException, Unit] = FileChannel.open(Path("foo.txt"))
+  .use { fileChannel =>
+    // fileChannel is only valid in this lexical scope
+    useChannel(fileChannel)
+  }
+```
+
+In the above example, the `FileChannel` will be opened and then provided to the function passed to `use`. The channel will always be closed when the `use` function completes, regardless of whether the operation succeeds, fails, dies or is interrupted. As long as the channel is only used within the function passed to `use`, then we're guaranteed not to have leaks.
+
+## Flexible Resource Scoping
+
+Sometimes there are situations where `ZManaged#use` is too limiting, because the resource lifecycle needs to extend beyond a lexical scope. An example of this is registering channels with a `Selector`. How can we do this using `ZManaged` while still avoiding the possibility of leaks? One way is to use [the "scope" feature of `ZManaged`][zio-scope].
+
+A scope is itself a managed resource. Other managed resources can be attached to a scope, which gives them the same lifecycle as the scope. When the scope is released, all the other resources that have been attached to it will also be released.
+
+```scala mdoc:silent
+ZManaged.scope.use { scope =>
+
+  val channel: IO[IOException, SocketChannel] = scope(SocketChannel.open).map {
+    case (_ /* early release */, channel) => channel
```

Comment:
You could use `case (earlyRelease @ _, channel)` here to give it a name but avoid an unused param compiler warning.

Reply:
Good trick, I didn't know that.